crypto/crypto/benches/noise.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Don't forget to run this benchmark with AES-NI enable.
+//! You can do this by building with the following flags:
+//! `RUSTFLAGS="-Ctarget-cpu=sandybridge -Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3"`.
```

Comment:
not sure how it works underneath, but that's the flags they give out in the aes-gcm doc 

I'm also not sure why we have to build with all these target features either :D I'll try to take a look

Reply:
Using `-C target-cpu=native` builds for your specific setup, so it may only run on your machine. On the other hand, `-C target-cpu=sandybridge ...` will run on any Intel cpu >= sandybridge that also supports the given SIMD extensions. However, we have to enable these compile flags everywhere we build libra (cargo x, all the dockerfiles, ...), which kinda sucks. If we just do the lazy thing and throw a `RUSTFLAGS=..` everywhere, it would limit available targets (e.g., we compile some client for iOS. i think some AWS vm's don't support it either).

cc @bmwill do you have any ideas on how we could make this ergonomic? maybe `cargo x` uses the simd features by default and has a flag to disable? then we have a `cargo x build` that we use in all the Dockerfiles?

Maybe a more sustainable approach is to make a wrapper around `aes` crate (maybe upstream? call it `aes-rt`) and do runtime dispatching using [`std::is_x86_feature_detected!("aes")`](https://doc.rust-lang.org/std/macro.is_x86_feature_detected.html) between the `aesni` and `aes-soft` crates. For example, it looks like `ring` does this kind of runtime dispatching, though it's in the C layer. Although, from a brief scan in the `RustCrypto` issues, it looks like this might be difficult to do correctly and efficiently.